iceoryx_utils/test/moduletests/test_cxx_list.cpp

```diff
@@ -0,0 +1,2283 @@
+
+    EXPECT_THAT(cTor, Eq(2));
+    EXPECT_THAT(customCTor, Eq(1));
+    EXPECT_THAT(dTor, Eq(3));
+    EXPECT_THAT(sut11.size(), Eq(0));
+    EXPECT_THAT(cnt, Eq(3));
+}
+
+TEST_F(list_test, writeContentViaDereferencedIterator)
+{
+    constexpr uint64_t TEST_VALUE{356u};
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_front(j);
+    }
+
+    auto sut1{sut};
+    auto iter = sut1.begin();
+    TestListElement element{TEST_VALUE};
+    *iter = element;
+    EXPECT_THAT(sut1.front().m_value, Eq(TEST_VALUE));
+}
+
+TEST_F(list_test, invalidIteratorErase)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    sut.erase(iter);
+
+    EXPECT_DEATH(sut.erase(iter), "");
+}
+
+TEST_F(list_test, invalidIteratorIncrement)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    sut.erase(iter);
+
+    EXPECT_DEATH(++iter, "");
+}
+
+TEST_F(list_test, invalidIteratorDecrement)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    sut.erase(iter);
+
+    EXPECT_DEATH(--iter, "");
+}
+
+TEST_F(list_test, invalidIteratorComparison)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    auto iter2 [[gnu::unused]] = sut.erase(iter);
+
+    EXPECT_DEATH(dummyFunc(sut.cbegin() == iter), "");
+}
+
+TEST_F(list_test, invalidIteratorComparisonUnequal)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    auto iter2 = sut.erase(iter);
+
+    EXPECT_DEATH(dummyFunc(iter2 != iter), "");
+}
+
+TEST_F(list_test, invalidIteratorDereferencing)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    auto iter2 [[gnu::unused]] = sut.erase(iter);
+
+    EXPECT_DEATH(dummyFunc((*iter).m_value), "");
+}
+
+TEST_F(list_test, invalidIteratorAddressOfOperator)
+{
+    for (uint64_t i = 0; i < TESTLISTCAPACITY; ++i)
+    {
+        const uint64_t j{i};
+        sut.emplace_back(j);
+    }
+
+    auto iter = sut.cbegin();
+    ++iter;
+    auto iter2 [[gnu::unused]] = sut.erase(iter);
+
+    EXPECT_DEATH(dummyFunc(iter->m_value == 12), "");
+}
+
+TEST_F(list_test, ListIsCopyableViaMemcpy)
+{
+    uint64_t i = 0;
+    using TestFwdList = list<TestListElement, TESTLISTCAPACITY>;
+    alignas(alignof(TestFwdList)) uint8_t* otherSutPtr = new uint8_t[sizeof(TestFwdList)];
```

Comment:
I am not sure this does what you expect. I think it will align the pointer itself, not the pointee (!). The memory returned by new will not be aligned. This will fail in weird ways when not handled correctly. One way is to allocate more memory (size + align) via new or malloc and align it afterwards with std::align. But also, otherSut could be allocated on the stack as a ```alignas(alignof(TestFwdList)) uint8_t buffer[sizeof(...)];``` Then take a pointer from this and use it as before, just no delete at the end but possibly a dtor call since this is somewhat like a placement new.

Reply:
Thx bringing this up. Addressed~!